improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -630,31 +662,32 @@ def test_old_naming_convention(self):
         diagnostic = "probability_of_rainfall_rate_above_threshold"
         threshold = AuxCoord(0.03, units="mm hr-1")
         result = plugin.construct_extract_constraint(diagnostic, threshold, True)
-        expected = (
-            "iris.Constraint("
-            "name='probability_of_rainfall_rate_above_threshold', "
-            "threshold=lambda cell: 0.03 * {t_min} < cell < 0.03 * "
-            "{t_max})".format(
-                t_min=(1.0 - WeatherSymbols().float_tolerance),
-                t_max=(1.0 + WeatherSymbols().float_tolerance),
-            )
-        )
-        self.assertIsInstance(result, str)
-        self.assertEqual(result, expected)
+        self.assertIsInstance(result, iris.Constraint)
+        self.assertTrue("threshold" in result.__dict__["_coord_values"])
 
     def test_zero_threshold(self):
         """Test construct_extract_constraint when threshold is zero."""
         plugin = WeatherSymbols()
-        diagnostic = "probability_of_rainfall_rate_above_threshold"
-        threshold = AuxCoord(0.0, units="mm hr-1")
+        diagnostic = "probability_of_number_of_lightning_flashes_per_unit_area_in_vicinity_above_threshold"
+        threshold = AuxCoord(0.0, units="m-2")
         result = plugin.construct_extract_constraint(diagnostic, threshold, False)
-        expected = (
-            "iris.Constraint("
-            "name='probability_of_rainfall_rate_above_threshold', "
-            "rainfall_rate=lambda cell:  -1e-12 < cell < 1e-12)"
+        expected = iris.Constraint(
+            name="probability_of_number_of_lightning_flashes_per_unit_area_in_vicinity_above_threshold",
+            number_of_lightning_flashes_per_unit_area=lambda cell: np.isclose(
+                cell.point,
+                threshold.points[0],
+                rtol=0,
+                atol=WeatherSymbols().float_abs_tolerance,
+            ),
+        )
+        self.assertIsInstance(result, iris.Constraint)
+        print(result)
+        print(self.cubes)
+        print(self.cubes.extract(result))
+        print(self.cubes.extract(expected))
```

Comment:
Need to get rid of these `print` statements.

Reply:
Fixed, sorry was left over from testing!